### PR TITLE
Remove `get_*_operator` functions, simplify commoncrawl logic

### LINKED ISSUE #242: [Improvement] Use operators directly, remove `get_.*_operator` functions
## Current Situation

We seem to have a lot of `get_<x>_operator` functions in this repository, some of which are only used once (and frequently are defined in the same DAG definition file).

## Suggested Improvement

We should see which of these we can remove and define the operators directly within the DAG context.

E.g.

```python
def get_my_operator(dag):
    return PythonOperator(
        task_id="my_task",
        python_callable=other_func,
        op_args=[arg1],
        depends_on_past=False,
        dag=dag,
    )

with dag:
    get_my_operator(dag)
```

Could be

```python
with dag:
    PythonOperator(
        task_id="my_task",
        python_callable=other_func,
        op_args=[arg1],
        depends_on_past=False,
        dag=dag,
    )
```

## Benefit

Reduced code & boilerplate within DAG definition files.

## Additional context

Some of these functions may be used to generate operators for tests, and some may be used to create reusable operators. We should assess the former to see if those tests are necessary, and we should likely keep the latter.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #242 by @AetherUnbound

## Description

As described in #242, many of the `get_*_operator` functions are only used once and create an unnecessary level of indirection when trying to view a DAG definition. This PR attempts to remove all `get_*_operator` functions and replace them with operator definitions directly within the DAG definition.

The commoncrawl DAGs were the most complex to refactor, and I ended up creating a utility function to house some of the utilities that were defined in the `operators.py` file initially.

While attempting to verify this I also found that our `.airflowignore` file was a bit too permissive and was preventing the commoncrawl DAGs from being parsed. I corrected this behavior and the DAGs parsed just fine.

## Testing Instructions

`just test` and `just run airflow dags list`, specifically checking for the presence of commoncrawl DAGs!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/commoncrawl/commoncrawl_etl.py [hunk 1]
```diff
@@ -136,7 +148,9 @@
                     "--master",
                     "yarn",
                     EXTRACT_SCRIPT_S3,
-                    "--default",
+                    # This was "--default" previously but a task within the DAG
+                    # modified it on DAG parse time to be this value.
+                    CC_INDEX_TEMPLATE,
```
Comment (zackkrida) [inline openverse_catalog/dags/commoncrawl/commoncrawl_etl.py:153]: Nice cleanup!

File: openverse_catalog/dags/commoncrawl/commoncrawl_utils.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+
+def load_file_to_s3(local_file, remote_key, bucket, aws_conn_id):
+    s3 = S3Hook(aws_conn_id=aws_conn_id)
```
Comment (zackkrida) [inline openverse_catalog/dags/commoncrawl/commoncrawl_utils.py:5]: ❤️

**Review by sarayourfriend — APPROVED** [review]
LGTM!

Here's the testing output:

```
#############################################################################
                         RUNNING "airflow dags list"
#############################################################################
dag_id                    | filepath                | owner          | paused
==========================+=========================+================+=======
airflow_log_cleanup       | maintenance/airflow_log | data-eng-admin | True  
                          | _cleanup_workflow.py    |                |       
brooklyn_museum_workflow  | providers/brooklyn_muse | data-eng-admin | True  
                          | um_workflow.py          |                |       
check_new_smithsonian_uni | providers/check_new_smi | data-eng-admin | True  
t_codes_workflow          | thsonian_unit_codes_wor |                |       
                          | kflow.py                |                |       
cleveland_museum_workflow | providers/cleveland_mus | data-eng-admin | True  
                          | eum_workflow.py         |                |       
commoncrawl_etl_workflow  | commoncrawl/commoncrawl | data-eng-admin | True  
                          | _etl.py                 |                |       
europeana_ingestion_workf | providers/europeana_ing | data-eng-admin | True  
low                       | estion_workflow.py      |                |       
europeana_workflow        | providers/europeana_wor | data-eng-admin | True  
                          | kflow.py                |                |       
finnish_museums_workflow  | providers/finnish_museu | data-eng-admin | True  
                          | ms_workflow.py          |                |       
flickr_ingestion_workflow | providers/flickr_ingest | data-eng-admin | True  
                          | ion_workflow.py         |                |       
flickr_workflow           | providers/flickr_workfl | data-eng-admin | True  
                          | ow.py                   |                |       
freesound_workflow        | providers/freesound_wor | data-eng-admin | True  
                          | kflow.py                |                |       
image_expiration_workflow | database/image_expirati | data-eng-admin | True  
                          | on_workflow.py          |                |       
jamendo_workflow          | providers/jamendo_workf | data-eng-admin | True  
                          | low.py                  |                |       
metropolitan_museum_workf | providers/metropolitan_ | data-eng-admin | True  
low                       | museum_workflow.py      |                |       
museum_victoria_workflow  | providers/museum_victor | data-eng-admin | True  
                          | ia_workflow.py          |                |       
nypl_workflow             | providers/nypl_workflow | data-eng-admin | True  
                          | .py                     |                |       
oauth2_authorization      | oauth2/authorize_dag.py | data-eng-admin | True  
oauth2_token_refresh      | oauth2/token_refresh_da | data-eng-admin | True  
                          | g.py                    |                |       
phylopic_workflow         | providers/phylopic_work | data-eng-admin | True  
                          | flow.py                 |                |       
rawpixel_workflow         | providers/rawpixel_work | data-eng-admin | True  
                          | flow.py                 |                |       
recreate_audio_popularity | database/recreate_audio | data-eng-admin | True  
_calculation              | _popularity_calculation |                |       
                          | .py                     |                |       
recreate_image_popularity | database/recreate_image | data-eng-admin | True  
_calculation              | _popularity_calculation |                |       
                          | .py                     |                |       
refresh_all_audio_popular | database/refresh_all_au | data-eng-admin | True  
ity_data                  | dio_popularity_data.py  |                |       
refresh_all_image_popular | database/refresh_all_im | data-eng-admin | True  
ity_data                  | age_popularity_data.py  |                |       
refresh_audio_view_data   | database/refresh_audio_ | data-eng-admin | True  
                          | view_data.py            |                |       
refresh_image_view_data   | database/refresh_image_ | data-eng-admin | True  
                          | view_data.py            |                |       
science_museum_workflow   | providers/science_museu | data-eng-admin | True  
                          | m_workflow.py           |                |       
smithsonian_workflow      | providers/smithsonian_w | data-eng-admin | True  
                          | orkflow.py              |                |       
staten_museum_workflow    | providers/statens_museu | data-eng-admin | True  
                          | m_workflow.py           |                |       
stocksnap_workflow        | providers/stocksnap_wor | data-eng-admin | True  
                          | kflow.py                |                |       
sync_commoncrawl_workflow | commoncrawl/sync_common | data-eng-admin | True  
                          | crawl_workflow.py       |                |       
tsv_to_postgres_loader    | database/loader_workflo | data-eng-admin | True  
                          | w.py                    |                |       
walters_workflow          | providers/walters_workf | data-eng-admin | True  
                          | low.py                  |                |       
wikimedia_commons_workflo | providers/wikimedia_wor | data-eng-admin | True  
w                         | kflow.py                |                |       
wikimedia_ingestion_workf | providers/wikimedia_ing | data-eng-admin | True  
low                       | estion_workflow.py      |                |       
wordpress_workflow        | providers/wordpress_wor | data-eng-admin | True  
                          | kflow.py                |                |       
                                                                             
```

I see two commoncrawl entries. Is that all that should be there?

**Comment by AetherUnbound** [comment]
> I see two commoncrawl entries. Is that all that should be there?

That's correct!

**Pull request merged**